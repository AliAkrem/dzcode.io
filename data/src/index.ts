import express from "express";
import { join } from "path";

import { getConfig } from "./config";
import { getCollection } from "./get/collection";
import { getEntry } from "./get/entry";

const app = express();

// Apply headers and logger
app.use((req, res, next) => {
  console.log(req.url);
  res.header("Access-Control-Allow-Origin", "*");
  next();
});

// Collections
app.get("/:type/:collection(\\S+.c.json$)", (req, res) => {
  const collection = getCollection(
    join(__dirname, ".."),
    req.params.type,
    req.params.collection,
    req.query.language as string,
  );

  if (typeof collection === "number") {
    return res.sendStatus(404);
  }

  return res.json(collection);
});

// Entries
app.get("/:type/:entry([\\/\\S]+.json$)", (req, res) => {
  const entry = getEntry(
    join(__dirname, ".."),
    `${req.params.type}/${req.params.entry.slice(0, -5)}`,
    undefined,
    req.query.language as string,
  );

  if (typeof entry === "number") {
    return res.sendStatus(404);
  }
  return res.json(entry);
});

// Start the server
const { port, url } = getConfig();
app.listen(port, () => console.log(`Data server listening at ${url}`));
