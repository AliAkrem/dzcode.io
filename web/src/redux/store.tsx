import { configureStore } from "@reduxjs/toolkit";
import { PropsWithChildren, useState } from "react";
import { Provider as ReduxProvider, useDispatch, useSelector } from "react-redux";

import { projectsPageSlice } from "./slices/projects-page";
import { settingsSlice } from "./slices/settings";

const makeAppStore = () => {
  return configureStore({
    reducer: {
      settings: settingsSlice.reducer,
      projectsPage: projectsPageSlice.reducer,
    },
  });
};

export function StoreProvider({ children }: PropsWithChildren): JSX.Element {
  const [appStore] = useState(makeAppStore());

  return <ReduxProvider store={appStore}>{children}</ReduxProvider>;
}

export type AppState = ReturnType<ReturnType<typeof makeAppStore>["getState"]>;
type AppDispatch = ReturnType<typeof makeAppStore>["dispatch"];

export const useAppDispatch = useDispatch.withTypes<AppDispatch>();
export const useAppSelector = useSelector.withTypes<AppState>();
