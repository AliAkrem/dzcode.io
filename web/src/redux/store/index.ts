import { combineReducers, compose, configureStore } from "@reduxjs/toolkit";
import * as slicesImport from "src/redux/store/slices";

// -----------------------------------------------------------------------------
export type Slices = typeof slicesImport;
export type SlicesKey = keyof typeof slicesImport;
export type Reducers = { [K in keyof Slices]: Slices[K]["reducer"] };
export type Actions = { [K in keyof Slices]: Slices[K]["actions"] };
export type ActionTypesRecord = {
  [K in keyof Actions]: `${keyof Actions[K] & string}`;
};
export type ActionType = {
  [K in keyof Actions]: `${K}/${keyof Actions[K] & string}`;
}[SlicesKey];
export type State = ReturnType<typeof rootReducer>;
// -----------------------------------------------------------------------------

export const slices = slicesImport;

const reducers = (Object.keys(slicesImport) as SlicesKey[]).reduce(
  (pV, sliceKey) => ({ ...pV, [sliceKey]: slicesImport[sliceKey].reducer }),
  {} as Reducers,
);
const rootReducer = combineReducers(reducers);

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export const store = configureStore({ reducer: rootReducer, enhancers: composeEnhancers });
