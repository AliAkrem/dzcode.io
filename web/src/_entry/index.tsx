import * as Sentry from "@sentry/browser";
import { Integrations } from "@sentry/tracing";
import { render } from "react-dom";
import { Provider } from "react-redux";
import { Router } from "react-router-dom";
import { store } from "src/redux";
import { getEnv } from "src/utils";
import { history } from "src/utils/history";

import { App } from "./app";

const env = getEnv();

if (env !== "development") {
  Sentry.init({
    dsn: "https://b42ee25f06c4416dac83de4ab8772e26@o953637.ingest.sentry.io/5902610",
    integrations: [new Integrations.BrowserTracing()],
    tracesSampleRate: 1.0,
    environment: env,
    debug: env !== "production",
    release: `web@${window.bundleInfo.version}`,
  });
}

render(
  <Provider store={store}>
    <Router history={history}>
      <App />
    </Router>
  </Provider>,
  document.getElementById("app-container"),
);
