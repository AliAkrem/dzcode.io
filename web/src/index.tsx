// Sentry initialization should be imported first!
import "src/utils/setup-sentry";

import { StrictMode } from "react";
import { createRoot } from "react-dom/client";

import { AppWithProviders } from "./_entry/app";

const rootEl = document.getElementById("root");
if (!rootEl) throw new Error("Root element not found!");

const root = createRoot(rootEl);
root.render(
  <StrictMode>
    <AppWithProviders />
  </StrictMode>,
);
