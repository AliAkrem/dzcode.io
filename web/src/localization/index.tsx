/* eslint sort-imports:off */
import PropTypes from "prop-types";
import { FC } from "react";
import { IntlProvider } from "react-intl";
import { useSelector } from "react-redux";
import { StateInterface } from "src/apps/main/redux";

import ar from "./locals/ar";
import en from "./locals/en";
import fr from "./locals/fr";

export const Localization: FC = (props) => {
  const locals: { [key: string]: any } = { fr, ar, en };

  const { settings } = useSelector<StateInterface, StateInterface>((state) => state);

  return (
    <IntlProvider
      messages={locals[settings.language.code]}
      locale={settings.language.code}
      defaultLocale="en"
    >
      {props.children}
    </IntlProvider>
  );
};

Localization.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Localization;
