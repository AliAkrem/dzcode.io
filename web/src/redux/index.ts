import { combineReducers, compose, configureStore } from "@reduxjs/toolkit";
import * as slices from "src/redux/reducers";

// -----------------------------------------------------------------------------
export type Slices = typeof slices;
export type SlicesKey = keyof typeof slices;
export type Reducers = { [K in keyof Slices]: Slices[K]["reducer"] };
export type Actions = { [K in keyof Slices]: Slices[K]["actions"] };
export type ActionTypesRecord = {
  [K in keyof Actions]: `${keyof Actions[K] & string}`;
};
export type ActionType = {
  [K in keyof Actions]: `${K}/${keyof Actions[K] & string}`;
}[SlicesKey];
export type State = ReturnType<typeof rootReducer>;
// -----------------------------------------------------------------------------

// @TODO-ZM: state getter where it chases the state when no action has been fired
// @TODO-ZM: enforce not using useSelector

const reducers = (Object.keys(slices) as SlicesKey[]).reduce(
  (pV, sliceKey) => ({ ...pV, [sliceKey]: slices[sliceKey].reducer }),
  {} as Reducers,
);
const rootReducer = combineReducers(reducers);

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export const createStore = () =>
  configureStore({ reducer: rootReducer, enhancers: composeEnhancers });

// @TODO-ZM: don't export store change it from `const` to `let`, and only export createStore(), which creates a store, returns it and also assign it to let store.
export const store = createStore();

export const actions = (Object.keys(slices) as SlicesKey[]).reduce(
  (pV, sliceKey) => ({
    ...pV,
    [sliceKey]: Object.keys(slices[sliceKey].actions).reduce(
      (contextualActions, actionName) => ({
        ...contextualActions,
        [actionName]: (...args: any[]) =>
          store.dispatch((slices[sliceKey].actions as any)[actionName](...args)),
      }),
      {},
    ),
  }),
  {} as Actions,
);
