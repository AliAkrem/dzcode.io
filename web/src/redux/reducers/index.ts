import { combineReducers } from "redux";

import { articles } from "./articles";
import { articlesPage } from "./articles-page";
import { contributePage } from "./contribute-page";
import { documentation } from "./documentation";
import { faqPage } from "./faq-page";
import { footerComponent } from "./footer-component";
import { learnPage } from "./learn-page";
import { navbarComponent } from "./navbar-component";
import { projectsPage } from "./projects-page";
import { settings } from "./settings";
import { teamPage } from "./team-page";

export const mainReducer = combineReducers({
  documentation,
  learnPage,
  articles,
  articlesPage,
  projectsPage,
  settings,
  faqPage,
  navbarComponent,
  footerComponent,
  contributePage,
  teamPage,
});
