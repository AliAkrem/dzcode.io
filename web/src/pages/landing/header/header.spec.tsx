import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import { createStore } from "src/redux";

import { Header } from "./index";

describe("header component", () => {
  test("should render header component", () => {
    const mainStore = createStore();

    const { container } = render(
      <Provider store={mainStore}>
        <MemoryRouter>
          <Header />
        </MemoryRouter>
      </Provider>,
    );

    expect(container).toMatchSnapshot();
  });
});
