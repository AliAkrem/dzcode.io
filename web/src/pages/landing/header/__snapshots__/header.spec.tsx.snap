// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header component should render header component 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-7"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-center"
      style="width: 100%; height: 100%;"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-7"
      >
        <div
          class="MuiBox-root MuiBox-root-8 makeStyles-left-2"
        >
          <h1
            class="MuiTypography-root makeStyles-title-3 MuiTypography-body1"
          >
            <span>
              <span
                class="makeStyles-greenHighlight-1"
              >
                Open-Source
              </span>
               Algerian Comunity
            </span>
          </h1>
          <p
            class="MuiTypography-root makeStyles-description-6 MuiTypography-body1 MuiTypography-colorTextPrimary"
          >
            <span>
              DzCode i/o helps you find, 
              <a
                href="/Contribute"
              >
                contribute
              </a>
               and add to the 
              <a
                href="/Projects"
              >
                list
              </a>
               of open-source projects that solve Algerian problems.
            </span>
          </p>
          <div
            class="MuiBox-root MuiBox-root-9"
          >
            <a
              class="makeStyles-callToAction-4"
              href="/Contribute"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-5 MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Make a Contribution
                </span>
              </button>
            </a>
            <a
              href="/FAQ"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-5 MuiButton-textPrimary MuiButton-textSizeLarge MuiButton-sizeLarge MuiButton-disableElevation"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Do you have a question?
                </span>
              </button>
            </a>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-5"
      />
    </div>
  </div>
</div>
`;
